test/core/libp2p/muxer/yamux/yamux_frame_test.cpp

```diff
@@ -0,0 +1,116 @@
+/**
+ * Copyright Soramitsu Co., Ltd. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include "libp2p/muxer/yamux/yamux_frame.hpp"
+
+#include <gtest/gtest.h>
+
+using namespace libp2p::muxer;
+using namespace kagome::common;
+
+class YamuxFrameTest : public ::testing::Test {
+ public:
+  static constexpr size_t data_length = 6;
+  static constexpr Yamux::StreamId default_stream_id = 1;
+  static constexpr uint32_t default_ping_value = 337;
+
+  Buffer data{0x12, 0x34, 0x45, 0x67, 0x89, 0xAB};
+
+  /**
+   * Check that all frame's fields are as expected
+   */
+  void checkFrame(std::optional<YamuxFrame> frame_opt, uint8_t version,
+                  YamuxFrame::FrameType type, YamuxFrame::Flag flag,
+                  Yamux::StreamId stream_id, uint32_t length,
+                  const Buffer &frame_data) {
+    ASSERT_TRUE(frame_opt);
+    auto frame = *frame_opt;
+    ASSERT_EQ(frame.version_, version);
+    ASSERT_EQ(frame.type_, type);
+    ASSERT_EQ(frame.flag_, flag);
+    ASSERT_EQ(frame.stream_id_, stream_id);
+    ASSERT_EQ(frame.length_, length);
+    ASSERT_EQ(frame.data_, frame_data);
+  }
+};
+
+TEST_F(YamuxFrameTest, ParseFrameSuccess) {
+  Buffer data_frame_bytes = dataMsg(default_stream_id, data);
+  auto frame_opt = parseFrame(data_frame_bytes.toVector());
+  checkFrame(frame_opt, YamuxFrame::kDefaultVersion,
```

Comment:
If you execute some function with asserts in test, please add a line with [scoped trace](https://github.com/google/googletest/blob/master/googletest/docs/advanced.md#adding-traces-to-assertions) with meaningful message (e.g. name of the test)

Reply:
Please apply the same logic for other tests when needed